doc/reference/vm_live_migration_internals.md

```diff
@@ -0,0 +1,96 @@
+---
+discourse: 50734
+---
+
+(vm-live-migration-internals)=
+# LXD VM live migration
+
+LXD supports {ref}`live-migration-vms` of virtual machines by streaming instance state from a source {abbr}`QEMU (Quick Emulator)` to a target QEMU. VM live migration is supported for all storage pool types.
+
+API extension: `migration_vm_live`
+
+## Conceptual process
+
+The live migration workflow varies depending on the type of storage pool used. The two key scenarios are non-shared storage and shared storage within a cluster (e.g., Ceph). If live state transfer is not supported by a target, a stateful stop is performed prior to migration.
+
+### Live migration for non-shared storage
+
+This process leverages the QEMU built-in Network Block Device (NBD) client-server mechanism to transfer the virtual machine's disk and memory state. Below is an overview of the steps:
+
+1. Set {abbr}`QMP (QEMU Machine Protocol)` capabilities.
+2. Temporary write buffer: Create a temporary {abbr}`QCOW2 (QEMU Copy on Write)` file to store writes that occur in the guest during the initial transfer of the root disk.
+3. Snapshot root disk.
```

Comment:
"Snapshot" confuses me a bit here. Is it when the `qcow2` gets a backing device and then becomes the VM's disk?

Reply:
Here, snapshot is a point-in-time snapshot of the root disk - this is what gets transferred. At this point, there are no writes happening to the root disk, only the `qcow2` file.

Reply:
I don't think snapshot is the write word to use here, I can see how it is confusing given the other steps.

Reply:
I've moved the "high-level steps" into this section, and gone into depth on how `migrateSendLive` works in the "Implementation details>call stack>`migrateSendLive`" section.

Reply:
I was thinking of adding pseudocode for each key function in the call stack, but I'm not sure if it's necessary. What do you think?

Reply:
Most of the action is happening in `migrateSendLive`.